### PR TITLE
Added arrow icon on collapsed submenu

### PR DESCRIPTION
**Pull request by rthera7**
Signed-off-by: Rakesh Kumar Thera <rthera@launchcg.com>

#### :heavy_check_mark: Checklist

Not Applicable

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/src/layout/Sidebar/Items.tsx [hunk 1]
```diff
@@ -83,134 +80,122 @@ export type SidebarItemClassKey =
...
+        marginLeft: -props.sidebarConfig.selectedIndicatorWidth,
       },
     }),
-    { name: 'BackstageSidebarItem' },
-  );
-
-// This is a workaround for this issue https://github.com/mui/material-ui/issues/15511
-// The styling of the `selected` elements doesn't work as expected when using a prop callback.
-// Don't use this pattern unless needed
-function useMemoStyles(sidebarConfig: SidebarConfig) {
```
Comment (jhaals) [inline packages/core-components/src/layout/Sidebar/Items.tsx:206]: Are these changes really intentional? We recently added this as a mitigation to a bug in mui4

**Review by jhaals — COMMENTED** [review]
Any particular reason why #10940 was closed in favour of this one? There seems to still be unrelated changes including incompatible style changes

File: packages/core-components/src/layout/Sidebar/Items.tsx (hunk 1 above)
Comment (rthera7) [reply packages/core-components/src/layout/Sidebar/Items.tsx:206]: No, I don't understand why these changes rolled in my commit, The changes I intended to be merged are lines 344-355 and 365 only

Comment (rthera7) [reply packages/core-components/src/layout/Sidebar/Items.tsx]: Hi jhaals, I have fixed the unwanted roll ins, to only the needed intentional changes, Thanks
